components/definition/plots/NetPyNEInclude.js

```diff
@@ -0,0 +1,367 @@
+import React, { Component } from 'react';
+import Menu from 'material-ui/Menu';
+import Paper from 'material-ui/Paper';
+import Divider from 'material-ui/Divider';
+import MenuItem from 'material-ui/MenuItem';
+import TextField from 'material-ui/TextField';
+import FlatButton from 'material-ui/FlatButton';
+import Popover from 'material-ui/Popover/Popover';
+import SelectField from 'material-ui/SelectField';
+import DropDownMenu from 'material-ui/DropDownMenu';
+import ContentAdd from 'material-ui/svg-icons/content/add';
+import ArrowDropRight from 'material-ui/svg-icons/navigation-arrow-drop-right';
+import Utils from '../../../Utils';
+import NetPyNEField from '../../general/NetPyNEField';
+
+var PythonControlledCapability = require('../../../../../js/communication/geppettoJupyter/PythonControlledCapability');
+
+
+export default class NetPyNEInclude extends Component {
+ 
+  constructor(props) {
+    super(props);
+    this.state = {
+      include: this.getDataTemplate(),
+      mainPopoverOpen: false,
+      label: ''
+    };
+  };
+ 
+  componentDidMount() {
+    this.collectInfo();
+    // this.updateLayout();
+  };
+ 
+  XOR = (a, b) => {
+    return ( a || b ) && !( a && b );
+  }
+  
+  getDataTemplate = () => {
+    return {
+      gids: [],
+      groups: [],
+      popids: {},
+      exclusive: false 
+    }
+  }
+  checkEqual = (a, b) => {
+    // compare if 2 shallow dicts are equal (no DOM, no 2nd lvl, no functions)
+    var aProps = Object.getOwnPropertyNames(a);
+    var bProps = Object.getOwnPropertyNames(b);
+    if (aProps.length != bProps.length) {
+        return false;
+    }
+    for (var i = 0; i < aProps.length; i++) {
+        var propName = aProps[i];
+        if (a[propName] !== b[propName]) {
+            return false;
+        }
+    }
+    return true;
+  }
+  
+  whoIsIncluded = (include, data) => {
```

Comment:
Can you please include some comments here? It is a bit of a nightmare to understand what is going on here...